tests/test_consumer.py

```diff
@@ -656,3 +656,55 @@ def test_full_consume_osci_example_2(mock_get_session, namespace, expected):
 
     assert all_expected_data == \
         json.loads(mock_requests.post.call_args_list[0][1]['data'])
+
+
+@mock.patch('resultsdbupdater.utils.retry_session')
+def test_full_consume_compose_msg(mock_get_session):
+    mock_rv = mock.Mock()
+    mock_rv.status_code = 201
+    mock_requests = mock.Mock()
+    mock_requests.post.return_value = mock_rv
+    mock_get_session.return_value = mock_requests
+    fake_msg_path = path.join(json_dir, 'compose_message.json')
+    with open(fake_msg_path) as fake_msg_file:
+        fake_msg = json.load(fake_msg_file)
+
+    assert consumer.consume(fake_msg) is True
+    # Verify the URLs called
+    assert mock_requests.post.call_args_list[0][0][0] == \
+        'https://resultsdb.domain.local/api/v2.0/results'
+    # Verify the post data
+    assert mock_requests.post.call_count == 1
+    url = "https://rtt-jenkins/job/compose-RHEL-X.0-rel-eng-tier2-acceptance/1/"
+    expected_data = {
+        "testcase": {
+            "name": "unknown.tier2.functional",
```

Comment:
You should probably address the fact that we are getting `unknown` here. Are you using an old message that doesn't have `namespace` set?

Reply:
Nice catch! Adding the missing (required) "namespace" attribute.